site/docs/glossary.md

```diff
@@ -0,0 +1,537 @@
+
+A data structure for collecting data on transitive dependencies. Optimized so
+that merging depsets is time and space efficient, because it’s common to have
+very large depsets (e.g. hundreds of thousands of files). Implemented to
+recursively refer to other depsets for space efficiency reasons. Rule
+implementations should not "flatten" depsets by converting them to lists unless
+the rule is at the top level of the build graph. Flattening large depsets incurs
+huge memory consumption. Also known as *Nested Sets* in Bazel's internal
+implementation.
+
+#### Disk cache
+
+A local on-disk blob store for the remote caching feature. Can be used in
+conjunction with an actual remote blob store.
+
+#### Distdir
+
+A read-only directory containing files that Bazel would otherwise fetch from the
+internet using repository rules. Enables builds to run fully offline.
+
+#### Dynamic execution
+
+An execution strategy that combines both the local and remote execution
+strategy, and through various heuristics, uses the execution result of the
+faster successful strategy. Certain [*actions*](#action) are better executed
+locally (e.g. linking), and others remotely (highly parallelizable compilation),
+so the dynamic execution strategy provides the best possible incremental and
+full build times.
+
+#### Execution phase
+
+The third phase of a build. Executes the [*actions*](#action) from the action
+graph created during the analysis phase. Executed actions invoke executables
+(compilers, scripts) to read and write [*artifacts*](#artifact). *Spawn
+strategies* control how these actions are executed: locally, remotely,
+dynamically, sandboxed, docker, and so on.
+
+#### Execution root
+
+A directory in the workspace’s output base directory where local actions are
+executed in non-sandboxed builds. The directory contents are mostly symlinks of
+input [*artifacts*](#artifact) from the workspace. The execution root also
+contains symlinks to external repositories as other inputs and the `bazel-out`
+directory to store outputs. Prepared during the loading phase by planting a
+"symlink forest" of the directories representing transitive closure of packages
+depended on by the target(s) to build. Accessible with `bazel info
+execution_root` on the command line.
+
+#### File
+
+See [*Artifact*](#artifact).
+
+#### Hermeticity
+
+The property of a build or test where there are no external influences on the
+build or test. This helps to enable deterministic and correct results. This
+usually means disallowing network access to actions, only allowing access to
+declared input [*artifacts*](#artifact), making timestamps and timezones fixed,
+restricting access/resetting environment variables, and using fixed seeds for
+random number generators.
+
+#### Incremental build
+
+A build that reuses the results of earlier builds to reduce build time and
+resource needs. Bazel aims to guarantee correct results for incremental builds.
+This is the opposite of a clean build.
```

Comment:
Maybe a few adjustments along the lines of:

"An incremental build reuses the results of earlier builds to reduce build time and resource usage. Dependency checking and caching aim to produce correct results for this type of build. An incremental build is the opposite of a clean build."